# Initial Data Loading and Variable Check

Import libraries

In [7]:
import pandas as pd
import numpy as np

Read data

In [8]:
responses = pd.read_csv("../../data/raw/counterfactual_responses_raw.csv")
metadata = pd.read_csv("../../data/raw/metadata_raw.csv")

In [9]:
print(responses.shape)
print(metadata.shape)

(336, 9)
(56, 19)


## Variables' Description

**metadata**

56 rows X 19 columns

- EXP: experiment 1 (self-focus or actors); experiment 2 (other-focus or readers).

- n

- Participant_ID: composite key made of EXP and n.

- Age

- Gender: 1 = female, 0 = male.

- Kno_level: knowledge level of ESL content the participant have studied byt the time they took the questionnaire according to the European Framework. Levels 1 to 9 mapping A1, A21, A22, B11, B12, B21, B22, C1 and C2.

- Per_level: participant's perceived level of ESL that could either match the participant's knowledge level or not. Levels 1 to 9 mapping A1, A21, A22, B11, B12, B21, B22, C1 and C2.
- Bilingual: whether the participant was raised as a simultaneous L2- English bilingual. 1 = Yes, 0 = No.

- Bi_Parents: some of the participants could be originally from Catalunya or Valencia; this variable looked into whether the participant was raised by bilingual parents. 1 = Yes, 0 = No.

- Multi_Kno: the participant studied other languages apart from L2-English. 1 = Yes, 0 = No.

- Multi_Spk: the participant speaks other languages apart from L2-English. 1 = Yes, 0 = No.

- Immersion: the participant took part in an L2-English immersion program. 1 = Yes, 0 = No.

- Prof_Need: the participant needs L2-English for professional reasons. 1 = Yes, 0 = No.

- Prof_Use: the participant speaks L2-English at work. 1 = Yes, 0 = No.

- Motivation: participant's motivation to keep learning L2 -English. 1 to 6, 6 = very motivated.

- Agent: the counterfactual's antecedent (if-clause) used as background has an explicit agent. 1 = Yes, 0 = No.

- Distractor: count of correct distractor answers. 1 to 6, 1 per subjunctive block.

- GN_Score: MacMillan's Upper-Intermediate to Advance online test used as control. Only those participants who scored more than 20 points were included in the data. Values range from 21 to 40.

- Difficulty: difficulty found in subjunctive blocks upon test completion. 1 to 10, 10 = very difficult.


**responses**

336 rows x 9 columns

- Participant_ID: composite key made of EXP and n.

- Focus: "I"  for EXP 1 or self-focus; "They" for EXP 2 or other-focus.

- Agent: the counterfactual's antecedent (if-clause) used as background has an explicit agent. 1 = Yes, 0 = No.

- Goal_Type: type of goal. 3 values: goal_frequent, goal_non_frequent, no_goal.

- Item: Item1 and Item 2, 2 items per type of goal. 

- Item_string: string matching the instance of counterfactual/ 3rd Conditional sentence used as background.

- Item_len: length of the string used as background.

- Response_tag: target desiderative sentence. Categories are correct_3, L1_transfer_3, L2_other_3. Tags are numbers as these responses are part of a bigger data set that looks into L2-English syntactic vs. replacement subjunctive performance differences in Spanish adults.

- Response: target desiderative sentence used for analysis and modelling. Categories are correct, L1_transfer, L2_other. Missing values were left in the set. 

    - correct: correct response option. 1 in 4 options.

    - L1_transfer: the response is incorrect but matches the expected incorrect response a Spanish ESL student would choose. 1 in 4 options.

    - L2_other: the response is incorrect but explores other L2-English alternatives. 2 in 4 options. Both options are collapsed into the same column. 
    

In [10]:
metadata.head()

,EXP,n,Participant_ID,Age,Gender,Kno_Level,Per_Level,Bilingual,Bi_Parents,Multi_Kno,Multi_Spk,Immersion,Prof_Need,Prof_Use,Motivation,Agent,Distractor,GN_Score,Difficulty
0,1,1,EXP1_01,25,0,6,7,0,1,0,1,0,1,1,4,1,4,23,3
1,1,2,EXP1_02,38,0,6,7,0,0,1,1,0,0,0,2,1,6,27,8
2,1,3,EXP1_03,27,1,4,5,0,1,1,1,0,1,1,6,1,4,28,7
3,1,4,EXP1_04,53,0,7,9,0,0,1,1,0,1,1,5,1,5,39,1
4,1,5,EXP1_05,53,1,7,8,0,0,1,1,1,1,1,4,1,6,35,1


In [11]:
responses.head()

,Partcipant_ID,Focus,Agent,Goal_Type,Item,Item_string,Item_len,Response_tag,Response
0,EXP1_01,I,1,no_goal,Item1,"""If I had heard the rainstorm, I would have fe...",11,L1_transfer_3,L1_transfer
1,EXP1_01,I,1,no_goal,Item2,"""If I had climbed the mountain cliff, I would ...",12,L1_transfer_3,L1_transfer
2,EXP1_01,I,1,goal_non_frequent,Item1,"""If I had created an online platform, I would ...",14,L1_transfer_3,L1_transfer
3,EXP1_01,I,1,goal_non_frequent,Item2,"""If I had invested in technology, I would have...",13,correct_3,correct
4,EXP1_01,I,1,goal_frequent,Item1,"""If I had saved money, I would have gone on ho...",11,correct_3,correct
